app/controllers/idv/cancellations_controller.rb

```diff
@@ -113,9 +113,8 @@ def session_go_back_path
       end
     end
 
-    def cancel_in_person_enrollment_if_exists
+    def cancel_establishing_in_person_enrollments
       return if !IdentityConfig.store.in_person_proofing_enabled
-      current_user.pending_in_person_enrollment&.update(status: :cancelled)
       UspsInPersonProofing::EnrollmentHelper.
         cancel_stale_establishing_enrollments_for_user(current_user)
```

Comment:
It looks like you didn't need to check where the user came from (ie: Ready to Verify) after all because of this function. Before entering your password, the enrollment is establishing). Only after entering your password (before going to the PO) would the enrollment be pending.  (All other's status- this function does not support ie: passed, failed, expired, and cancelled) so that is why you renamed it. 

Also, method cancel_in_person_enrollment_if_exists is a dup - inside sessions_controller so further more I support this name change- each method is not exactly the same. Without testing it- I think the method in sessions_controller supports cancel your barcode. I will test on Monday to confirm.